### ISSUE TITLE
Compiling Altera's Atlas SOC design is failing

### ISSUE THREAD

**Issue by domibel** [dwesterg/atlas-soc-ghrd#1]
Instructions are not working.
https://rocketboards.org/foswiki/view/Documentation/AtlasSoCCompileHardwareDesign#3

mkdir atlas
cd atlas
wget https://github.com/dwesterg/atlas-soc-ghrd/tarball/master -O atlas-soc-ghrd.tgz
mkdir atlas-soc-ghrd
tar -xzvf atlas-soc-ghrd.tgz --strip-components 1 -C atlas-soc-ghrd
cd atlas-soc-ghrd
make all

This ends in a bug described here:
https://www.altera.com/support/support-resources/knowledge-base/solutions/rd05082015_486.html

The workaround doesn't work. The qsys-script in Quartus 16.0 is still crashing.

**Comment by dwesterg** [comment]
are you doing this with a full quartus install?

**Comment by domibel** [comment]
Yes on Ubuntu 16.04 with

Quartus-16.0.0.211-linux-complete.tar
Size: 24.0 GB MD5: 59F3A591F34A105C8BC616BDEC1F8977

**Comment by domibel** [comment]
I am running into the same issue on CentOS release 6.8 (Final) with
SoCEDSSetup-15.0.1.60-linux.run and
Quartus-15.0.2.153-linux-complete.tar

qsys-script --cmd="source scripts/create_ghrd_qsys_ATLAS_SOC_GHRD.tcl; build_qsys scripts/qsys_default_components.tcl ATLAS_SOC_GHRD" 2>&1 |
 tee logs/ATLAS_SOC_GHRD.qsys_gen.stamp.log
2016.08.23.13:23:46 Info: Doing: <b>qsys-script --cmd=source scripts/create_ghrd_qsys_ATLAS_SOC_GHRD.tcl; build_qsys scripts/qsys_default_co
mponents.tcl ATLAS_SOC_GHRD</b>
-- Accepted parameter $devkitname = ATLAS_SOC_GHRD
-- Accepted parameter $qsys_name = soc_system
-- Accepted parameter $devicefamily = CYCLONEV
-- Accepted parameter $device = 5CSEMA4U23C6
2016.08.23.13:23:52 Info: create_system soc_system
2016.08.23.13:23:52 Info: set_project_property DEVICE_FAMILY CYCLONEV
2016.08.23.13:23:52 Info: set_project_property DEVICE 5CSEMA4U23C6
2016.08.23.13:23:52 Info: add_instance hps_0 altera_hps     # here it stalls

Is the the Altera Cyclone V Atlas SOC still supported?
https://www.altera.com/products/boards_and_kits/dev-kits/partners/cyclonev_soc_atlas.html

**Comment by dwesterg** [comment]
I'm going to send it to you compiled, where should I send it.  I can't
reproduce your error.

Dalon

**Comment by domibel** [comment]
Thanks, but I am more interested in compiling the files myself. Other people are able to reproduce the bug as well.
https://forum.rocketboards.org/t/atlas-board-build-script-gets-stuck/232/7

Could you please describe your compiler environment? OS, Quartus version etc...

**Comment by dwesterg** [comment]
Sure, I rub this on either a centos 5 / 6, our Fedora 23 machine.  The only
time I have seen issues is with quartus web edition, or quartus installs
without the stratix family installed.

Dalon

**Comment by dwesterg** [comment]
Oh, quartus version, everything from 15.0 to 16.0.2

**Comment by domibel** [comment]
Ok, I can setup another environment to try to reproduce the build / bug.
In the meanwhile, please send the compiled files to domibel@debian.org .

**Comment by domibel** [comment]
I found the issue, the environment variable LM_LICENSE_FILE was not set, and the qsys-script doesn't complain. So this should now be reproducible on your end as well.

After setting LM_LICENSE_FILE correctly, the compilation succeeds without errors with Quartus 15.0.

I am getting a few errors with Quartus 16.0:
2016.08.23.14:21:38 Info: Doing: <b>qsys-script --cmd=source scripts/create_ghrd_qsys_ATLAS_SOC_GHRD.tcl; build_qsys scripts/qsys_default_components.tcl ATLAS_SOC_GHRD</b>
-- Accepted parameter $devkitname = ATLAS_SOC_GHRD
-- Accepted parameter $qsys_name = soc_system
-- Accepted parameter $devicefamily = CYCLONEV
-- Accepted parameter $device = 5CSEMA4U23C6
2016.08.23.14:21:43 Info: create_system soc_system
2016.08.23.14:21:43 Info: set_project_property DEVICE_FAMILY CYCLONEV
2016.08.23.14:21:43 Info: set_project_property DEVICE 5CSEMA4U23C6
2016.08.23.14:21:43 Info: add_instance hps_0 altera_hps 
ERROR: Device family can not be determined
ERROR: Device family can not be determined
ERROR: Device family can not be determined
2016.08.23.14:21:53 Info: set_instance_parameter_value hps_0 MEM_VENDOR Other
2016.08.23.14:21:54 Info: set_instance_parameter_value hps_0 MEM_FORMAT DISCRETE
2016.08.23.14:21:54 Info: set_instance_parameter_value hps_0 RDIMM_CONFIG 0000000000000000

Can I ignore them?

**Comment by dwesterg** [comment]
Yes, ignore those.  Been meaning to clean them up for a bit now.

Thanks,
Dalon
